plugins/module_utils/crypto/support.py

```diff
@@ -203,10 +203,16 @@ def load_certificate(path, content=None, backend='cryptography'):
     if backend == 'pyopenssl':
         return crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
     elif backend == 'cryptography':
-        try:
-            return x509.load_pem_x509_certificate(cert_content, cryptography_backend())
-        except ValueError as exc:
-            raise OpenSSLObjectError(exc)
+        if any([x in cert_content for x in [b'-----BEGIN CERTIFICATE-----', b'-----END CERTIFICATE-----']]):
```

Comment:
Please use `identify_pem_format()` from `ansible_collections.community.crypto.plugins.module_utils.crypto.pem`.

Also please add a new parameter to this function which allows to enable this feature (default must be to not enable it), and only use this in `x509_certificate_info`. Otherwise you are potentially breaking other code which relies on this function to not load DER certificates. (Also if `backend == 'pyopenssl'` when this feature is enabled, fail.)

Reply:
Thank you for your suggestion. I added the suggested changes.